lib/node_modules/@stdlib/blas/base/wasm/zscal/test/test.module.main.js

```diff
@@ -0,0 +1,295 @@
+	var mem;
+	var mod;
+
+	mem = new Memory({
+		'initial': 0
+	});
+	mod = new Module( mem );
+	t.strictEqual( mod.main.length, 4, 'returns expected value' );
+	t.end();
+});
+
+tape( 'a module instance has a `main` method which scales elements from `x` by `alpha`', function test( t ) {
+	var actualX;
+	var viewX;
+	var mem;
+	var mod;
+	var ap;
+	var xe;
+	var xp;
+
+	mem = new Memory({
+		'initial': 1
+	});
+	mod = new Module( mem );
+	mod.initializeSync();
+
+	xp = 0;
+	ap = 48;
+
+	mod.write( xp, new Complex128Array( [ 0.3, 0.1, 0.5, 0.0, 0.0, 0.5 ] ) );
+	mod.write( ap, new Float64Array( [ 0.4, -0.7 ] ) );
+
+	xe = new Float64Array( [ 0.19, -0.17, 0.2, -0.35, 0.35, 0.2 ] );
+
+	mod.main( 3, ap, xp, 1 );
+
+	actualX = new Complex128Array( 3 );
+	mod.read( xp, actualX );
+	viewX = new Float64Array( actualX.buffer );
+	isApprox( t, viewX, xe, 1.0 );
+
+	t.end();
+});
+
+tape( 'a module instance has a `main` method which supports specifying a `x` stride', function test( t ) {
+	var actualX;
+	var viewX;
+	var xbuf;
+	var mem;
+	var mod;
+	var ap;
+	var xe;
+	var xp;
+	var x;
+
+	mem = new Memory({
+		'initial': 1
+	});
+	mod = new Module( mem );
+	mod.initializeSync();
+
+	xp = 0;
+	ap = 96;
+
+	xbuf = new Float64Array([
+		0.1,  // 1
+		0.1,  // 1
+		3.0,
+		6.0,
+		-0.6, // 2
+		0.1,  // 2
+		4.0,
+		7.0,
+		0.1,  // 3
+		-0.3, // 3
+		7.0,
+		2.0
+	]);
+	x = new Complex128Array( xbuf.buffer );
+
+	mod.write( xp, x );
+	mod.write( ap, new Float64Array( [ 0.4, -0.7 ] ) );
+
+	xe = new Float64Array([
+		0.11,  // 1
+		-0.03, // 1
+		3.0,
+		6.0,
+		-0.17, // 2
+		0.46,  // 2
+		4.0,
+		7.0,
+		-0.17, // 3
+		-0.19, // 3
+		7.0,
+		2.0
+	]);
+
+	mod.main( 3, ap, xp, 2 );
+
+	actualX = new Complex128Array( 6 );
+	mod.read( xp, actualX );
+	viewX = new Float64Array( actualX.buffer );
+	isApprox( t, viewX, xe, 3.0 );
+
+	t.end();
+});
+
+tape( 'a module instance has a `main` method which supports specifying a negative `x` stride', function test( t ) {
+	var actualX;
+	var viewX;
+	var xbuf;
+	var mem;
+	var mod;
+	var ap;
+	var xe;
+	var xp;
+	var x;
+
+	mem = new Memory({
+		'initial': 1
+	});
+	mod = new Module( mem );
+	mod.initializeSync();
+
+	xp = 0;
+	ap = 96;
+
+	xbuf = new Float64Array([
+		0.1,  // 3
+		0.1,  // 3
+		3.0,
+		6.0,
+		-0.6, // 2
+		0.1,  // 2
+		4.0,
+		7.0,
+		0.1,  // 1
+		-0.3, // 1
+		7.0,
+		2.0
+	]);
+	x = new Complex128Array( xbuf.buffer );
+
+	mod.write( xp, x );
+	mod.write( ap, new Float64Array( [ 0.4, -0.7 ] ) );
+
+	xe = new Float64Array([
+		0.11,  // 3
+		-0.03, // 3
+		3.0,
+		6.0,
+		-0.17, // 2
+		0.46,  // 2
+		4.0,
+		7.0,
+		-0.17, // 1
+		-0.19, // 1
+		7.0,
+		2.0
+	]);
+
+	mod.main( 3, ap, xp, -2 );
+
+	actualX = new Complex128Array( 6 );
+	mod.read( xp, actualX );
+	viewX = new Float64Array( actualX.buffer );
+	isApprox( t, viewX, xe, 3.0 );
+
+	t.end();
+});
+
+tape( 'a module instance has a `main` method which returns a reference to the input array', function test( t ) {
```

Comment:
It doesn't return a `reference`. It returns a `pointer`. See the Wasm implementation for `daxpy`.